### ISSUE TITLE
feat: Support "empty" enum variant

### ISSUE THREAD

**Issue by yuvalsw** [starkware-libs/cairo#3452]
Support an enum variant of type unit (`()`), without specifying the unit type. For example:
```
enum MyEnum {
   A, // Currently we need "A: (),"
   B: usize
}
```

**Comment by neotheprogramist** [comment]
I will take that

**Comment by neotheprogramist** [comment]
In, Rust a ```unit``` type "()" is not equivalent to 'no type'. Here:
```rust
enum MyEnum {
    Empty,
    Void(),
    Unit(()),
}

fn main() {
    let a = MyEnum::NoType;
    match a {
        MyEnum::Empty => (),
        MyEnum::Void() => (),
        MyEnum::Unit(()) /* or MyEnum::Unit(_)  */ => (),
    }
}
```
all 3 variants have/hold different type. [Nothing, empty-tuple, tuple holding empty-tuple]
My understanding is that the second type (Void()) is hard to achieve in the current Cairo syntax. 

Suppose the following Cairo declaration:
```
enum MyEnum {
   Empty,
   Unit: ()
}
```

I see two solutions:
1. The fields' types are equivalent and the difference is merely syntactic and Empty being a shorthand for Unit.
2. The fields' types differ with ```Empty``` having no type and ```Unit``` holding ```unit``` type (more Rust like).

Is there (or might be) a Cairo-specific reason that forces (assuming a reasonable implementation effort) a particular solution?

**Comment by neotheprogramist** [comment]
@ArielElp Could you take a look at that?

**Comment by yuvalsw** [comment]
Thanks for raising this, Paweł.
In Cairo we have a single type per variant (which could be a tuple), thus it's a bit simpler than Rust. We'd want `Empty` variant to be a syntactic sugar of `Unit` variant, according to your example. So in your example they both hold the same value - a unit type.

Your suggested solution 1 is the one we are looking for.
Let me know if you have further questions, and thanks for contributing!

**Comment by neotheprogramist** [comment]
Fixed by #3569
